src/mlpack/methods/adaboost/adaboost_main.cpp

```diff
@@ -202,8 +202,10 @@ int main(int argc, char *argv[])
     else if (weakLearner == "perceptron")
       m.WeakLearnerType() = AdaBoostModel::WeakLearnerTypes::PERCEPTRON;
 
+    const size_t numClasses = arma::max(labels) + 1;
```

Comment:
Do we require continuous labels starting with 0? If not I think this doesn't return the correct number of labels.